src/values/ECMAScriptFunctionValue.js

```diff
@@ -0,0 +1,43 @@
+/**
+ * Copyright (c) 2017-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+/* @flow */
+
+import type { Realm } from "../realm.js";
+import type { BabelNodeBlockStatement, BabelNodeSourceLocation, BabelNodeLVal } from "babel-types";
+import { FunctionValue, ObjectValue } from "./index.js";
+import * as t from "babel-types";
+
+/* ECMAScript Function Objects */
+export default class ECMAScriptFunctionValue extends FunctionValue {
+  constructor(realm: Realm, intrinsicName?: string) {
+    super(realm, intrinsicName);
+  }
+
+  $ThisMode: string;
```

Comment:
While you are at it, we can make the type of $ThisMode more restrictive. Looks like it (eventually) must be "lexical" | "strict" | global"